src/transformers/pipelines.py

```diff
@@ -457,7 +457,11 @@ def _parse_and_tokenize(self, *args, pad_to_max_length=True, **kwargs):
         # Parse arguments
         inputs = self._args_parser(*args, **kwargs)
         inputs = self.tokenizer.batch_encode_plus(
-            inputs, add_special_tokens=True, return_tensors=self.framework, pad_to_max_length=pad_to_max_length,
+            inputs,
+            add_special_tokens=True,
+            return_tensors=self.framework,
+            pad_to_max_length=pad_to_max_length,
+            max_length=self.tokenizer.max_len,
```

Comment:
This is the key change.

Reply:
I don't think we want that

Reply:
I think we should throw an error message here if the encoded input is longer than `self.tokenizer.max_len` - see issue #4501 for this.